src/LayerEditor/ui/data/layer_model.py

```diff
@@ -83,4 +88,20 @@ def save(self, geo_model: LayerGeometrySerializer, region_id_to_idx: dict):
                     top_type,
                     self.top_top,
                     bottom_type,
-                    bottom_top)
\ No newline at end of file
+                    bottom_top)
+
+    def set_region_to_selected_shapes(self, region: Region):
+        """Sets regions of shapes only in this layer."""
+        for g_item in self.block.selection._selected:
+            dim = g_item.dim
+            if self.is_stratum:
+                dim += 1
+            if dim == region.dim or region.dim == -1:
+                self.set_region_to_shape(g_item, region)
+
+    def get_shape_region(self, g_item: [GsPoint, GsSegment, GsPolygon]) -> Region:
```

Comment:
This seems to me more elegant. I would have to pass shape_id and dimension, because shape_ids are unique only for shapes of the same dimension. It is also convenient for debuging. Lastly when there are ids everywhere it is very constraining. Therefore I would like to get rid of as much ids as possible, or at least to hide them.

Reply:
Changed to only passing dim and shape_id. Done